### ISSUE TITLE
Creating demographics and labor outcomes for MORG-may data

### ISSUE THREAD

**Issue by dgentil1** [dgentil1/Katrina#20]

**Comment by dgentil1** [comment]
Guys, I've been working on the monthly files and I've revisited Borjas and Peri's papers and I've decided to take a second look to the merged outgoing rotation group sample (the other sample that Peri uses in his paper apart from March CPS and that last year we tried to use but you told me it was not possible to unify the codes to our codes). It turns out that the morg files are a much better sample for our purposes and I started to work on them. 

1. I was able to homogenize the MSA's codes to our unified metcode2.
2. I added a cpi99 index analogous to the one in ASEC data.
3. Created a do file that list per metcode2 the treat, treat_expanded and control and use that to assign treatment status in the morg data.

Now, I thought that is a much better use of Mar's time to do what the title of the task describes and stop the other tasks. I will reassign them to Daniele and I, because results from this data are much more reliable and less noisy as explained by several papers including the one by Peri. Therefore, this is lkely to make a substantia chunk of the paper. Basically now we have to create all variables that we need in this data in a way analogous to what we did for ASEC data. 

Steps: 
1. Download data [from here](http://www.nber.org/morg/annual/). 
2. Take a look to the data documentation [here](http://www.nber.org/morg/docs/cpsx.pdf)
3. Download from [here](https://fred.stlouisfed.org/series/CPIAUCSL) the file that you will need to input into M_cpi_morg to create the cpi99 that is already created in M_preclean_morg (both M_preclean_morg and M_cpi_morg are already committed)
4. Start working in creating the labor outcomes and demographics in M_preclean_morg.
5. If something doesn't look immediately possible check how Peri did it, because he compared ASEC and Morg in his paper. His code and data can be found [here](http://giovanniperi.ucdavis.edu/data-for-peri-and-yasenov.html) and the paper [here](http://giovanniperi.ucdavis.edu/uploads/5/6/8/2/56826033/mariel_may_10_2016.pdf)
6. If something still looks like is not possible to reproduce analogously let us know and we will think about a way to adapt it or we might hange the way we do it in ASEC. 

Thanks!

**Comment by marinabosque** [comment]
So, am I in charge of this and also the couple of things we said yesterday???????

**Comment by dgentil1** [comment]
Yes, if you finish too fast then you and keep going with something else. But note that shortening the term of the DID is done now

**Comment by marinabosque** [comment]
Check the new branch --> !!

Things to comment about the **D_Precleaning_morg.do**:

1. At the beginning of the dofile there is a part to **keep** just the variables we are using. In this way the dataset is less heavy and runs faster. It is very likely that we have to add variables for the analysis we are going to carry out later on. Just add them. The ones needed for the preclean and idkatrina are already there.

2. **Ethnicity variable**: Notice that to create this variable we use (as Peri does) race and ethnic variables. The ethnic variable is coded as:
Mexican American
Chicano
Mexicano 
Puerto Rican
Cuban
Central or South American
Other Spanish
All other
Don’t know

And from the race we are just using (as Peri):
White 
Black
[...]
because in the ethnic variable there are some coded as "don't know"or "." that have one of the two codes in the race variable. The other races are gonna be considered "Others".

2a) **Mexican American** are mexican or white?
2b) A central or south american can be considered black? jajajaj I have considered NOT.
2c) There are some missing "don't know" after recoding the variable (16 people if I don't remember bad). As we did in the CPS (if there were some), they are coded as **"Others"**. Am I right in doing this?

3. **White and Blue collar variables**: 

3a) Notice that in the "D_Precleaning_asec.do" the variables manuf, blue and white collars and, consequently, workcat are NOT well coded. Manuf and blue collar overlap and therefore the categorical variable that we create afterwards makes no sense... So, **my proposal is to get rid of manufacturing and divide only between blue and white collar** (unless the variable manuf is essential in the SCM). This is what I did for the MORG. If you agree on that I can take care of changing the variable in ASEC as well.

3b) Related to this variable, I have a doubt: do we consider **Armed Forces as blue or white collar**? There is controversy in Internet, so I guess we can make our own judgement. I would bet for blue collar.

4. **Wages variable**: 

4a) Notice that there are available **hours worked, hourly and weekly wages** in MORG. As far as I can think about, there is **no way to get the yearly wages**. We can assume the weeks that an individual works in a year and then multiply the hours worked per week by that number to get the total hours worked in one year. Then, multiply this by the hourly wage to get the yearly wage. Otherwise we can assume the weeks worked per year and multiply this number by the weekly earnings. What do you guys prefer? Or maybe you can think about another solution.

4b) **TOP-CODING for the weekly wage**. If you check the [data documentation](http://www.nber.org/morg/docs/cpsx.pdf) and you look at the Wages part, it says: > [...] But check earnwke (this is the variable for weekly wages) for top-coding [...].

Then when checking the description of the variable it doesn't say anything about top-coding (and it does for other earning variables): > Edited or computed earnings per week in this job. Includes overtime tips and commissions. For hourly workers, computed Item 25a times Item 25c appears here. For weekly workers, edited Item 25d appears here. Also for 1989 on, there are no zero values, suggesting an undocumented change in universe. For 1979-1988 this is from locations 417-419.

By looking at Peri code, he does this: 

gen uearnhre = (earnwke/uhourse)
drop earnhr* uhourse 

global vars uearnhre earnwke
foreach i of global vars {
	replace `i' = `i' * 1.3775 if year == 1985 
	replace `i' = `i' * 1.3513 if year == 1986 
	replace `i' = `i' * 1.3046 if year == 1987 
	replace `i' = `i' * 1.2532 if year == 1988
	replace `i' = `i' * 1.1959 if year == 1989
	replace `i' = `i' * 1.1344 if year == 1990 
	replace `i' = `i' * 1.0886 if year == 1991
	replace `i' = `i' * 1.0564 if year == 1992
	replace `i' = `i' * 1.0259 if year == 1993 
	replace `i' = `i' * 1 if year == 1994 
	replace `i' = `i' * 0.9727 if year == 1995 
	replace `i' = `i' * 0.9449 if year == 1996
	replace `i' = `i' * 0.9234 if year == 1997
	replace `i' = `i' * 0.9093 if year == 1998 
	replace `i' = `i' * 0.8898 if year == 1999
	replace `i' = `i' * 0.8608 if year == 2000
	replace `i' = `i' * 0.8372 if year == 2001
	replace `i' = `i' * 0.8241 if year == 2002
	replace `i' = `i' * 0.8056 if year == 2003
	replace `i' = `i' * 0.7846 if year == 2004
	gen log`i' = log(`i') 
}

I have the feeling that the loop is for the cpi94.

So, **my guess is that we don't have to correct for the top-coding**.

5. **Household income** and **poverty variable** not available... Any suggestion?

6. Do we need the individual identifier later on?

7. I have reordered the dofile according to the CPS one and put them "beautiful".

8. I have created the dofile for the identification of Katrina and the main file that collects the programs for datafixing of MORG.

9. I have corrected some mistakes there were in the ASEC files like the cap saveold or some labels that were wrongly coded (workcat).

Things to comment about the **D_cpi_morg.do**:

10. Maybe I am a super ignorant with this question (and now it will be remembered forever with this Github jajajajaj), but why just keeping May?


11. I have an additional question about the **D_Idkatrina_asec.do** (and will be also for the MORG) that I am wondering. It was the comment that there was in the dofile and Diego ask us to remove. Can be commented in another moment!


If there isn't anything else urgent, I will try to help Daniele with his tasks from now until Friday. But feel free to assign me more things if they have to be ready on Friday!! I can keep working on the katrina identification in MORG dofile if you think it is worthy!

Thanks guys!! And sorry for the long explanation, we can do an Skype call to solve it if better.

**Comment by marinabosque** [comment]
Oh I got the thing of May....It's MORG may jajajajajaj Forget about point 10

**Comment by marinabosque** [comment]
Check the new branch --> Correcting-ASEC-and-adding-MORG!!

Sorry for the delay, the occupation and industry variables were more tricky to recode as expected.

Things to comment about the **D_Precleaning_morg.do** and **D_Precleaning_asec.do**:
1. **Variable occupation** added and recoded. Also **variable industry** divided into blue, white collar, and "don't know". I don't know which one do you wanna use, so I put both. 

By the way, to make things consistent, I finally leave the armed forces outside of any of the two categories. There were some variables for which "don't know" and armed forced had the same code, not possible to split.

[I had to download the data because we didn't have the variable occ in ASEC]

2. I haven't erased yet poverty and hh_income. Alternatives variables in MORG can be: number of kids, union membership, full/part time job, marital status, duration of unemployment... Not very interesting actually...

*********************
3. I have added the **D_Idkatrina_morg.do** with the corrections Diego was telling in the previous comment. Just a little comment. To merge the pre-treatment labor outcome variables, I had to create a new dataset (that is the one merged to the complete dataset, called 'pre_laboroutcomes.dta'). Couldn't keep the same as in ASEC because we would have overwritten them.

*********************

By the way, I don't know what were you referring when you said: Please remove the old files then because otherwise when merging to master we will confuse github.
Were you referring to the McIntosh one?

**Comment by dgentil1** [comment]
Hola @marinabosque ! 

1. Sound good!

2. Let's keep those variables in the datasets for now and later when cleaning the code before submitting if we finally don't use them we can drop them. 

3. Sounds good, just add the weights that the data documentation suggests. 

Regarding your comment about id_katrina_morg:

I just realized that we have the Katrina evacuees variable in morg too! Then what we should do is repeat the whole process and see whether treatment assignment agrees in bot datasets!!!!! This is exciting actually :D The reason for this is that given that morg sample size is better we might get more precise assignment. If they completely agree then we gain credibility, and if they don't we will probably learn how to build a cleaner design with those that agree! Let me know after you are done with points 2. and 3. and I will open a task for myself to explore the tratment asignment in morg and if it coincides with the one in ASEC. 

Thanks!

**Comment by marinabosque** [comment]
Okey, ahora me pongo con esos dos puntos!

**Comment by marinabosque** [comment]
Hola amigos!!

2. I have just added the variables in MORG. I leave them not recoded. If we see we need them, I personally take care of cleaning them. 

3. Done!

Then, I have rewritten the **D_IDkatrina_morg.do** in the same way we have the one for ASEC in order to carry out the same analysis. I have also created the data set **D_Get_treat_control_morg.do**. I guess this will be useful to compare treatment and control assignment :)
